AngelProtocolFinance/angelprotocol-web-app
# RC-v1.9: wigdet fixes and refactors

Ticket(s):
fixes bugs:
- Non-unique Keys for tokens + need network name or something to tell same named tokens apart. Thinking to group them by network in the dropdown would be best?
- Selecting the endowment in the dropdown does not allow for URL generation
- No changes to "preview" widget when various options selected (ex. Hide text)

## Explanation of the solution
* separate overall widget config state from widget configurer form state
* organize files
<img width="480" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/5b5dfb0a-9a0e-4090-b171-fcf1eac724f6">  

* reduce markup depth and utilize grid layout  
* validate form input prior to generating snippet
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/53d054d0-5b97-4038-994f-2d4fd474f765)

* remove tokens field
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/a3447046-db3a-4f5c-b681-e9889ea9c581)


* changes to donation components to properly consume config from widget

### Tests 
NOTE: this is rendered on an isolated `html` with vs code [extension live server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) To duplicate,
- make sure web dev server is running
- create `html` file
- paste `iframe` snippet inside `<body/>`
- run html file via live server


- correctly renders config (hidden advanced options, description text hidden)
NOTE: borders and dimensions are set in `<iframe/>` tag
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/67d1ec37-eb9c-4606-ab1f-d5fd7d3577e8)



## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes
